crates/lib/src/compiler/quilc.rs

```diff
@@ -307,13 +307,25 @@ struct QuilToNativeQuilResponse {
     /// The compiled program
     quil: String,
     /// Metadata about the compiled program
+    #[serde(default)]
     metadata: Option<NativeQuilMetadata>,
 }
 
+#[allow(unused_qualifications)]
+fn deserialize_none_as_default<'de, D, T>(deserializer: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de> + std::default::Default,
+{
+    let opt = Option::deserialize(deserializer)?;
+    Ok(opt.unwrap_or_default())
+}
+
```

Comment:
Shouldn't need this. I think you just need to do `serde(default)` and it will automatically use the default.